control-board/src/motion/robot_controller.rs

```diff
@@ -229,4 +239,320 @@ impl<'a> BodyVelocityController<'a> {
+                    } else {
+                        reply_cmd.command_code = PCC_NACK_INVALID_TYPE_FOR_NAME;
+                        return Err(reply_cmd);
+                    }
+                },
+                // VEL_CGKF_ENCODER_NOISE => {
+
+                // },
+                // VEL_CGKF_GYRO_NOISE => {
+                    
+                // }
+                // VEL_CGKF_PROCESS_NOISE => {
+
+                // },
+                // VEL_CGFK_INITIAL_COVARIANCE => {
+
+                // },
+                // VEL_CGKF_K_MATRIX => {
+
+                // },
+                RC_BODY_VEL_LIMIT => {
+                    if param_cmd.data_format == VEC3_F32 {
+                        // write the new data, then read it back into the reply
+                        self.body_velocity_limit.as_mut_slice().copy_from_slice(unsafe { &param_cmd.data.vec3_f32 });
+                        unsafe { reply_cmd.data.vec3_f32.copy_from_slice(self.body_velocity_limit.as_slice()); }
+    
+                        reply_cmd.command_code = PCC_ACK;
+                    } else {
+                        reply_cmd.command_code = PCC_NACK_INVALID_TYPE_FOR_NAME;
+                        return Err(reply_cmd);
+                    }
+                },
+                RC_BODY_ACC_LIMIT => {
+                    if param_cmd.data_format == VEC3_F32 {
+                        // write the new data, then read it back into the reply
+                        self.body_acceleration_limit.as_mut_slice().copy_from_slice(unsafe { &param_cmd.data.vec3_f32 });
+                        unsafe { reply_cmd.data.vec3_f32.copy_from_slice(self.body_acceleration_limit.as_slice()); }
+    
+                        reply_cmd.command_code = PCC_ACK;
+                    } else {
+                        reply_cmd.command_code = PCC_NACK_INVALID_TYPE_FOR_NAME;
+                        return Err(reply_cmd);
+                    }
+                },
+                RC_WHEEL_ACC_LIMIT => {
+                    if param_cmd.data_format == VEC4_F32 {
+                        // write the new data, then read it back into the reply
+                        self.wheel_acceleration_limits.as_mut_slice().copy_from_slice(unsafe { &param_cmd.data.vec3_f32 });
+                        unsafe { reply_cmd.data.vec3_f32.copy_from_slice(self.wheel_acceleration_limits.as_slice()); }
+    
+                        reply_cmd.command_code = PCC_ACK;
+                    } else {
+                        reply_cmd.command_code = PCC_NACK_INVALID_TYPE_FOR_NAME;
+                        return Err(reply_cmd);
+                    }
+                },
+                _ => {
+                    defmt::debug!("unimplemented key write in RobotController");
+                    reply_cmd.command_code = PCC_NACK_INVALID_NAME;
+                    return Err(reply_cmd);
+                }
+            }
+        }
+
+        // match param_cmd.parameter_name {
```

Comment:
Is this code WIP or can be deleted?

Reply:
Wanna test my rewrite above first on actual hardware, then it's all old/dead.